usaspending_api/common/recipient_lookups.py

```diff
@@ -137,14 +137,19 @@ def as_sql(self, compiler, connection):
                 [],
             )
 
-    return queryset.annotate(**{field_name: RecipientId()})
+    return queryset.annotate(**{alias_name: RecipientId()})
 
 
-def annotate_recipient_id(field_name, queryset):
+def annotate_recipient_id(alias_name, queryset, id_lookup=None, parent_id_lookup=None, name_lookup=None):
+    if not all([v is not None for v in [id_lookup, parent_id_lookup, name_lookup]]):
```

Comment:
Does this mean that if someone uses this function with only one of the optional params set, it will be silently overwritten? Sounds dangerous to me.

Reply:
Good point. Would you prefer an error be thrown if one or two are present and not all three?

Reply:
I think that would be an upgrade. One of your famous data classes strikes me as better still.

Reply:
If all three are necessary for the SQL, why don't you remove the default values?

Reply:
It is either an all or nothing where the values of 
```
id_lookup = "{outer_table}.recipient_unique_id"
parent_id_lookup = "{outer_table}.parent_recipient_unique_id"
name_lookup = "{outer_table}.recipient_name"
``` 
make up the defaults without defining the long values in the function definition. I didn't want to break current functionality and so I added these as an optional way to overwrite the generated SQL in rare cases such as this.